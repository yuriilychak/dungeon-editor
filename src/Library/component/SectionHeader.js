import React, {Fragment, memo} from "react";
import Add from '@material-ui/icons/Add';
import CreateNewFolder from '@material-ui/icons/CreateNewFolder';

import ToolButton from "./ToolButton";
import Icon from "./Icon";
import {makeStyles} from "@material-ui/styles";

const useStyles = makeStyles({
    title: {
        userSelect: "none",
        flexGrow: 1,
        display: "flex",
        alignItems: "center",
        textAlign: "left",
        paddingLeft: 8,
        textOverflow: "ellipsis",
        whiteSpace: "nowrap",
        overflow: "hidden"
    }
});

const SectionHeader = ({
                           id,
                           icon,
                           titleText,
                           addElementText,
                           addDirectoryText,
                           onAddDirectory
                       }) => (
    <Fragment>
        <Icon name={`${icon}_root`}/>
        <div className={useStyles().title}>{titleText}</div>
        <ToolButton
            title={addElementText}
            Icon={Add}
            onClick={() => {}}
        />
        <ToolButton
            title={addDirectoryText}
            Icon={CreateNewFolder}
            onClick={() => onAddDirectory(id)}
        />
    </Fragment>
);


export default memo(SectionHeader);
