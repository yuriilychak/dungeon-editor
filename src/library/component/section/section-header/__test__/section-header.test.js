import React from "react";
import { createMount } from "@material-ui/core/test-utils";

import SectionHeader from "../section-header";

describe("SectionHeader test",() => {
    const onAddDirectory = jest.fn();
    const onAddFile = jest.fn();
    const wrapper = createMount()(
        <SectionHeader
            id={5}
            icon="test"
            titleText="test"
            addElementText="test"
            addDirectoryText="test"
            onAddFile={onAddFile}
            onAddDirectory={onAddDirectory}
        />
    );

    it("default snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Add file", () => {
        wrapper.find("button").at(0).simulate("click");
        expect(onAddFile).toHaveBeenCalled();
    });

    it("Add directory", () => {
        wrapper.find("button").at(1).simulate("click");
        expect(onAddDirectory).toHaveBeenCalled();
    });
});
