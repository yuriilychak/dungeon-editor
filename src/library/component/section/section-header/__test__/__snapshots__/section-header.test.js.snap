// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader test default snapshot 1`] = `"<div class=\\"library-section-header-title\\">test</div>"`;
